arches/app/datatypes/core/geojson_feature_collection.py

```diff
@@ -0,0 +1,434 @@
+                )
+        return {
+            "nodeid": node.nodeid,
+            "name": layer_name,
+            "layer_definitions": layer_def,
+            "icon": layer_icon,
+            "legend": layer_legend,
+            "addtomap": node.config["addToMap"],
+        }
+
+    def after_update_all(self, tile=None):
+        with connection.cursor() as cursor:
+            if tile is not None:
+                cursor.execute(
+                    "SELECT * FROM refresh_tile_geojson_geometries(%s);",
+                    [tile.pk],
+                )
+            else:
+                cursor.execute("SELECT * FROM refresh_geojson_geometries();")
+
+    def default_es_mapping(self):
+        mapping = {
+            "properties": {
+                "features": {
+                    "properties": {
+                        "geometry": {
+                            "properties": {
+                                "coordinates": {"type": "float"},
+                                "type": {"type": "keyword"},
+                            }
+                        },
+                        "id": {"type": "keyword"},
+                        "type": {"type": "keyword"},
+                        "properties": {"type": "object"},
+                    }
+                },
+                "type": {"type": "keyword"},
+            }
+        }
+        return mapping
+
+    def is_a_literal_in_rdf(self):
+        return True
+
+    def to_rdf(self, edge_info, edge):
+        # Default to string containing JSON
+        g = Graph()
+        if edge_info["range_tile_data"] is not None:
+            data = edge_info["range_tile_data"]
+            if data["type"] == "FeatureCollection":
+                for f in data["features"]:
+                    del f["id"]
+                    del f["properties"]
+            g.add(
+                (
+                    edge_info["d_uri"],
+                    URIRef(edge.ontologyproperty),
+                    Literal(JSONSerializer().serialize(data)),
+                )
+            )
+        return g
+
+    def from_rdf(self, json_ld_node):
+        # Allow either a JSON literal or a string containing JSON
+        try:
+            val = json.loads(json_ld_node["@value"])
+        except:
+            raise ValueError(
+                f"Bad Data in GeoJSON, should be JSON string: {json_ld_node}"
+            )
+        if "features" not in val or type(val["features"]) != list:
+            raise ValueError(f"GeoJSON must have features array")
+        for f in val["features"]:
+            if "properties" not in f:
+                f["properties"] = {}
+        return val
+
+    def validate_from_rdf(self, value):
+        if type(value) == str:
+            # first deserialize it from a string
+            value = json.loads(value)
+        return self.validate(value)
+
+    def pre_tile_save(self, tile, nodeid):
+        if tile.data[nodeid] == "":
+            tile.data[nodeid] = None
```

Comment:
This will allow `check_geojson_value` to run against `None` if the node value is already `None`.  So I think this check should simply be: 
```
if tile.data[nodeid]:
    tile.data[nodeid] = self.check_geojson_value(tile.data[nodeid])
```

Changing the value from and empty string to None should probably go in the datatype's `clean` method which gets called before this one on save.

The `clean` method gets called before this one, and this logic should actually be there.